### PR TITLE
[12.0] erplibre website snippets

### PR DESCRIPTION
**Pull request by carms-ng**
Add a suite of website snippets, from basic HTML building blocks to sectional structures.

These 4 modules composes of the following snippets
- h1..h6, p, small, img, avatar, ul, btn, pill, section header
- 3-step how it works, linked image grid, 5-part text panel
- flippable and extendable cards
- horizontal and vertical timelines

### CODE REVIEW

File: erplibre_website_snippets_basic_html/views/snippets.xml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      <t t-snippet="erplibre_website_snippets_basic_html.s_h6" t-thumbnail="/erplibre_website_snippets_basic_html/static/src/img/h6.png"/>
+      <t t-snippet="erplibre_website_snippets_basic_html.s_p" t-thumbnail="/erplibre_website_snippets_basic_html/static/src/img/p.png"/>
+      <t t-snippet="erplibre_website_snippets_basic_html.s_small" t-thumbnail="/erplibre_website_snippets_basic_html/static/src/img/small.png"/>
+      <t t-snippet="erplibre_website_snippets_basic_html.s_pill" t-thumbnail="/erplibre_website_snippets_basic_html/static/src/img/pill.png"/>
+    </xpath>
+  </template>
+
+  <!-- Add snippet_structure to menu -->
+  <template id="add_snippet_structure_to_menu" inherit_id="website.snippets">
+    <xpath expr="//div[@id='snippet_structure']/div[@class='o_panel_body']" position="inside">
```
Comment (mathben) [inline erplibre_website_snippets_basic_html/views/snippets.xml]: Error-prone use of @class in view add_snippet_structure_to_menu (): use the hasclass(*classes) function to filter elements by their classes

File: erplibre_website_snippets_cards/views/snippets.xml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    </section>
+  </template>
+
+  <!-- Add snippets to menu -->
+  <template id="add_snippet_structure_to_menu" inherit_id="website.snippets">
+    <xpath expr="//div[@id='snippet_structure']/div[@class='o_panel_body']" position="inside">
```
Comment (mathben) [inline erplibre_website_snippets_cards/views/snippets.xml]: Error-prone use of @Class in view add_snippet_structure_to_menu (): use the hasclass(*classes) function to filter elements by their classes

File: erplibre_website_snippets_structures/views/snippets.xml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            </a>
+          </div>
+        </div>
+      </div>
+    </section>
+  </template>
+
+  <!-- Add snippets to menu -->
+  <template id="add_snippet_structure_to_menu" inherit_id="website.snippets">
+    <xpath expr="//div[@id='snippet_structure']/div[@class='o_panel_body']" position="inside">
```
Comment (mathben) [inline erplibre_website_snippets_structures/views/snippets.xml]: Error-prone use of @Class in view add_snippet_structure_to_menu (): use the hasclass(*classes) function to filter elements by their classes

File: erplibre_website_snippets_timelines/views/snippets.xml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            <h4>Organization Inception</h4>
+          </div>
+        </div>
+      </div>
+    </section>
+  </template>
+
+  <!-- Add snippet to menu -->
+  <template id="add_snippet_structure_to_menu" inherit_id="website.snippets">
+    <xpath expr="//div[@id='snippet_structure']/div[@class='o_panel_body']" position="inside">
```
Comment (mathben) [inline erplibre_website_snippets_timelines/views/snippets.xml]: Error-prone use of @Class in view add_snippet_structure_to_menu (): use the hasclass(*classes) function to filter elements by their classes

File: erplibre_website_snippets_basic_html/views/snippets.xml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      <div class="container">
+        <h5 class="o_h5 pt8 pb4">Lorem ipsum dolor sit amet</h5>
+        <h2 class="o_h2 pt4 pb8">Lorem ipsum</h2>
+      </div>
+    </section>
+  </template>
+
+  <!-- Add snippet_content to menu -->
+  <template id="add_snippet_content_to_menu" inherit_id="website.snippets">
+    <xpath expr="//div[@id='snippet_content']/div[@class='o_panel_body']" position="inside">
```
Comment (mathben) [inline erplibre_website_snippets_basic_html/views/snippets.xml]: use [hasclass('o_panel_body')] instead of [@class='o_panel_body']

**Review by mathben — COMMENTED** [review]
About extendable cards, flippable cards
by default 3 instead of one.
Bug, when adding block of extend card, 1 appear in size of 1/3. When changing nb_column to 2, 2 appears in size of 1/2, when changing to 1, size if 1/1.
Same for flippable cards

About timeline, the horizontal, it has an offset at left and no offset at right.
Can we add 3 item by default, like the vertical timeline?

**Comment by carms-ng** [comment]
@mathben 

> About extendable cards, flippable cards
> by default 3 instead of one.

Done!

> Bug, when adding block of extend card, 1 appear in size of 1/3. When changing nb_column to 2, 2 appears in size of 1/2, when changing to 1, size if 1/1.
> Same for flippable cards

If you check odoo's original snippets "Three Columns", changing nb_col to 2 || 1 yields the same results.  It's not a bug.  Duplicating the cards yields better results.

> About timeline, the horizontal, it has an offset at left and no offset at right.

This is done by design.  It was originally full screen width, but I think it looks better to align the content equally on the left, but it wouldn't make sense to cut the timeline before it ends on the right.  

> Can we add 3 item by default, like the vertical timeline?

Done!

File: erplibre_website_snippets_basic_html/views/snippets.xml (hunk 1 above)
Comment (carms-ng) [reply erplibre_website_snippets_basic_html/views/snippets.xml]: updated to `div[hasclass('o_panel_body')]`

File: erplibre_website_snippets_structures/views/snippets.xml (hunk 3 above)
Comment (Carms-dev) [reply erplibre_website_snippets_structures/views/snippets.xml]: updated to `div[hasclass('o_panel_body')]`

File: erplibre_website_snippets_cards/views/snippets.xml (hunk 2 above)
Comment (Carms-dev) [reply erplibre_website_snippets_cards/views/snippets.xml]: updated to `div[hasclass('o_panel_body')]`

**Comment by carms-ng** [comment]
Edited all 4 commits to incorporate the updates accordingly

**Review by mathben — APPROVED** [review]
Good work, thanks for you contribution.

**Pull request merged**